server/routes/api/user.js

```diff
@@ -23,15 +23,16 @@ router.get('/favorite', async (req, res) => {
     const userId = req.user;
 
     const { rows } = await db.query(`
-    SELECT array_agg(favorite_ads_of_users.advertisement_id) as favorite_ads
+    SELECT array_agg(fav.advertisement_id) as favorite_ads
     FROM jobs.users users
-    LEFT JOIN jobs.favorite_ads_of_users favorite_ads_of_users ON favorite_ads_of_users.upn=$1
-    where users.upn=$1;`,
+    LEFT JOIN jobs.favorite_ads_of_users fav ON fav.upn=$1
```

Comment:
not every ser have favorite, that why you need a left join here. 
I need the where cloase to get only 1 user.
and change it to `fav ON fav.upn=users.upn` so that will be more clear